tests/cli_integration.rs

```diff
@@ -244,18 +245,18 @@ mod cli_integration {
 
 	#[test(tokio::test)]
 	async fn start_tls() {
-		let (_, server) = common::start_server(StartServerArguments {
+		let (_, mut server) = common::start_server(StartServerArguments {
 			tls: true,
 			wait_is_ready: false,
```

Comment:
Have you considered `wait_is_ready: true`?

(might have to change `common::start_server` to send the is-ready queries over TLS if `tls: true`).

Reply:
`wait_is_ready` is kinda playing opposite role. My main intention was to fully remove any timeouts from tests, like `std::thread::sleep(...)`. B/c at my laptop `start_tls` test has spurious failures.
The whole idea of this PR is to introduce different startup modes. The `Test` one for all the integration tests, where you just start the DB, start web server and immediately shutdown. And `Normal` mode when you start everything and start listening for socket connections.

Reply:
The idea of `wait_is_ready` is that the test will wait until the server starts, relaxing the current short timeout to a much longer timeout if needed.

In my personal opinion, adding startup modes to solve this problem introduces too much complexity (distracting from the property being tested) when `wait_is_ready` would do the job.

Reply:
Fair enough.